src/tests/Directory.Build.targets

```diff
@@ -0,0 +1,87 @@
+    <PackageDownload Include="TestAssets.Windows.x64.6.0"
+                     Version="[$(TestAssetsVersion)]" />
+    <PackageDownload Include="TestAssets.Windows.x86.6.0"
+                     Version="[$(TestAssetsVersion)]" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(NeedsTestAssets)' == 'true' and '$(BuildTests)' == 'true' and ('$(OS)' == 'Unix' OR '$(OS)' == 'Windows_NT')">
+    <PackageDownload Include="TestAssets.Linux.x64.6.0"
+                     Version="[$(TestAssetsVersion)]" />
+    <PackageDownload Include="TestAssets.Linux.arm64.6.0"
+                     Version="[$(TestAssetsVersion)]" />
+  </ItemGroup>
+
+  <!-- We need this for the binplacing for testing assets.
+  This should be removed at some point as it's brittle (harcodes versions and creates native-managed coupling). -->
+  <Target Name="_PublishPackageReferences"
+      AfterTargets="PostBuildEvent"
+      Condition="$(NeedsPublishing) == 'true'"
+      DependsOnTargets="$(_BeforePublishNoBuildTargets);$(_CorePublishTargets)" />
+
+  <Target Name="InvokeInstallRuntimes"
+      Condition="'$(IsUnitTestProject)' == 'true'"
+      BeforeTargets="RunTests;VSTest">
+      <MSBuild Projects="$(RepositoryEngineeringDir)InstallRuntimes.proj" Targets="InstallTestRuntimes" />
+  </Target>
+
+  <Target Name="WriteTestVersionManifest"
+          BeforeTargets="Build"
+          Inputs="$(VersionsPropsPath)"
+          Outputs="$(DebuggerVersionConfigFileName)">
+
+    <ItemGroup>
+      <TestConfigFileLines Include="Header">
+        <ConfigFileEntry>
+<![CDATA[
+<Configuration>
+  <InternalReleaseTesting>$(InternalReleaseTesting)</InternalReleaseTesting>
+  <PrivateBuildTesting>$(PrivateBuildTesting)</PrivateBuildTesting>
+]]>
+        </ConfigFileEntry>
+      </TestConfigFileLines>
+      <TestConfigFileLines Include="@(RuntimeTestVersions)">
+        <ConfigFileEntry>
+<![CDATA[
+  <RuntimeVersion%(RuntimeTestVersions.Identity)>%(RuntimeTestVersions.Runtime)</RuntimeVersion%(RuntimeTestVersions.Identity)>
+  <AspNetCoreVersion%(RuntimeTestVersions.Identity)>%(RuntimeTestVersions.AspNet)</AspNetCoreVersion%(RuntimeTestVersions.Identity)>
+  <TargetFramework%(RuntimeTestVersions.Identity)>%(RuntimeTestVersions.TargetFramework)</TargetFramework%(RuntimeTestVersions.Identity)>
+]]>
+        </ConfigFileEntry>
+      </TestConfigFileLines>
+      <TestConfigFileLines Include="End">
+        <ConfigFileEntry>
+<![CDATA[
+</Configuration>
+]]>
+        </ConfigFileEntry>
+      </TestConfigFileLines>
+    </ItemGroup>
+
+    <WriteLinesToFile File="$(DebuggerVersionConfigFileName)" Lines="@(TestConfigFileLines->Metadata('ConfigFileEntry'))" Overwrite="true" WriteOnlyWhenDifferent="true" />
+
+    <Message Importance="High" Text="Created config file $(DebuggerVersionConfigFileName)" />
+
+    <ItemGroup>
+      <FileWrites Include="$(DebuggerVersionConfigFileName)" />
+    </ItemGroup>
+  </Target>
+</Project>
```

Comment:
I wanted to comment on the *.UnitTest renames but GH won't let me add a comment there.

I'd suggest if we rename these projects we should use ".Tests" instead of ".UnitTests". Many of these tests are more accurately described as functional or integration tests rather than unit tests.

Reply:
Sounds good to me.